src/generators/genlua.ml

```diff
@@ -512,8 +564,14 @@ and gen_expr ?(local=true) ctx e = begin
 		spr ctx " end )({";
 		concat ctx ", " (fun (f,e) -> print ctx "%s = " (anon_field f); gen_value ctx e) fields;
 		spr ctx "})";
-	| TField (x,f) ->
-		gen_value ctx x;
+	| TField ({eexpr = TLocal tv} as e, f) when is_multireturn ctx e ->
+		(match f with
+		| FInstance(_,_,cf)->
+			let tmpname = multi_value_tmp_lookup tv.v_meta cf.cf_name in
```

Comment:
Here's where we look up the temporary name of a multireturn variable field, and use that instead of field access.